.github/workflows/pipeline.yml

```
@@ -0,0 +1,51 @@
+
+    - name: Publish Application
+      run: dotnet publish --configuration Release --output publish
+
+    - name: Upload Build Artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: build-output
+        path: publish
+
+    - name: Upload Test Coverage Report
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)